### PR TITLE
Prefer live Rework reset truth over stale Merging projection

### PR DESCRIPTION
**Pull request by Kbediako**
## Summary
- prefer live `Rework` reset truth over stale rehydrated `Merging` / PR lifecycle truth in provider observability and selected-run projection
- classify `issue-context` PR attachments into current vs historical vs conflicting vs unknown while deduplicating identical PR URLs and preserving ambiguous `Merging` truth
- add the CO-220 docs-first packet plus focused regressions for reset, fallback-only, duplicate-attachment, and ambiguous replacement-PR cases

## Testing
- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npm run build`
- `npm run lint` (3 pre-existing warnings in `orchestrator/tests/DelegationMcpHealth.test.ts`; no lint errors)
- `CODEX_VITEST_PROGRESS=1 npx vitest run --config vitest.config.core.ts`
- `npm run pack:smoke`
- `DIFF_BUDGET_OVERRIDE_REASON="CO-220 requires the mandated docs-first packet plus focused cross-surface regression coverage for provider projection, issue-context attachment truth, and the review-raised fallback edge cases; splitting further would reduce auditability more than it would reduce review risk." node scripts/diff-budget.mjs`
- focused regressions:
  - `npx vitest run orchestrator/tests/ProviderIssueObservability.test.ts orchestrator/tests/ProviderLinearWorkflowFacade.test.ts orchestrator/tests/SelectedRunProjection.test.ts`

## Review notes
- initial standalone reviews surfaced reset-suppression and attachment-classification bugs; those fixes are included here with added regressions
- the final bounded review rerun entered a repeated command-intent/read-only retry loop without reaching a new verdict, so I completed a manual correctness + elegance fallback review on the final diff instead of blocking handoff

## Summary by CodeRabbit

* **Bug Fixes**
  * Prioritised live Rework reset truth over stale rehydrated projections and ensured closed PR attachments are treated as historical while preserving genuine live merging and current PR truth.

* **Documentation**
  * Added PRD, technical spec, action plan, checklist and task docs with acceptance criteria, rollback guidance and validation steps.

* **Tests**
  * Added tests validating attachment classification, cache-reuse behaviour and projection suppression after Rework resets.

* **Chores**
  * Updated task registry and docs-freshness entries to track the new task and review cadence.

### CODE REVIEW

File: .agent/task/linear-27a61a75-236f-40c6-8cb0-8e4921708d91.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- [ ] `npm run lint`
+- [ ] `npm run test`
+- [ ] `npm run docs:check`
+- [ ] `npm run docs:freshness`
+- [ ] `npm run repo:stewardship`
+- [ ] `node scripts/diff-budget.mjs`
+- [ ] `FORCE_CODEX_REVIEW=1 npm run review -- --uncommitted` or manual fallback with recorded reason.
+- [ ] Explicit elegance/minimality pass recorded.
+- [ ] `npm run pack:smoke`
+
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply .agent/task/linear-27a61a75-236f-40c6-8cb0-8e4921708d91.md]: @coderabbitai Synced the task mirrors and recorded the concrete docs-review manifest path in `1ad86920c`, so the checklist state is now audit-accurate on the current branch.

File: orchestrator/src/cli/control/providerLinearWorkflowFacade.ts [hunk 2]
```diff
@@ -861,6 +881,255 @@ function shouldFallbackToCachedIssueContextFromWorkflowError(
...
+  const record = value as Record<string, unknown>;
+  return {
+    state: normalizeOptionalString(record.state as string | null | undefined),
+    merged_at: normalizeOptionalString(
+      (record.mergedAt ?? record.merged_at) as string | null | undefined
+    ),
+    updated_at: normalizeOptionalString(
+      (record.updatedAt ?? record.updated_at) as string | null | undefined
+    )
+  };
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/providerLinearWorkflowFacade.ts:985]: @coderabbitai Fixed in `1ad86920c`: `mapIssuePullRequestAttachmentSnapshot(...)` now rejects unrecognized payload objects whose normalized snapshot fields are all null, so they classify as `unknown` instead of falling through as non-terminal PR truth.

File: orchestrator/src/cli/control/providerLinearWorkflowFacade.ts [hunk 3]
```diff
@@ -3196,6 +3468,26 @@ function shouldReuseCachedIssueContextForRead(
...
+    return false;
+  }
+  if (issue.attachments.some((attachment) => parseGitHubPullRequestUrl(attachment.url))) {
+    return true;
+  }
+  return classifyProviderLinearWorkflowState({
+    state: issue.state?.name ?? null,
+    state_type: issue.state?.type ?? null
+  }).normalizedState === 'merging';
+}
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/providerLinearWorkflowFacade.ts:3502]: @coderabbitai Fixed in `1ad86920c`: `issueContextCacheRecordCarriesLivePrTruth(...)` now treats cached `Merging` state with historical-only PR attachments as live PR truth, and the facade regression suite now covers both structured cached current PR truth and the `Merging` historical-only cache path.

**Comment by Kbediako** [comment]
@coderabbitai Addressed the actionable review items on commit `1ad86920c`:
- rejected unrecognized PR snapshot payloads so they classify as `unknown`
- treat cached `Merging` state with historical-only PR attachments as live PR truth
- expanded the degraded-headroom cache regression coverage to include structured cached PR truth
- synced the task mirrors with the concrete docs-review manifest evidence

Please re-review when the current run catches up.

**Comment by Kbediako** [comment]
@coderabbitai Acknowledged the bounded-concurrency hydration nitpick from the completed re-review.

I’m not taking that change in CO-220 because this lane is correctness-only and a fanout change would alter GitHub request/rate-limit behavior without performance evidence. I filed follow-up CO-230 to evaluate bounded-concurrency PR snapshot hydration separately with explicit latency, rate-pressure, and failure-handling evidence.

For CO-220, the current branch keeps the reset-truth fix bounded and the review surfaces are otherwise green.

**Comment by Kbediako** [comment]
Final acknowledgement for the rerun after the CO-230 follow-up note: no new findings landed after the bot replay, so the branch stays unchanged.

CO-220 remains scoped to the reset-truth correctness fix, and CO-230 tracks the separate bounded-concurrency evaluation.

**Comment by Kbediako** [comment]
Superseded by the issue's reset to Rework. Closing this attempt and carrying the outstanding review findings into a fresh branch from origin/main for a new handoff.
